light-cli/src/commands/history/index.ts

```diff
@@ -0,0 +1,110 @@
+import { Command, Flags, ux } from "@oclif/core";
+import { BN } from "@coral-xyz/anchor";
+import { IndexedTransaction } from "@lightprotocol/zk.js";
+import { CustomLoader, getUser } from "../../utils/utils";
+
+type TransactionHistory = {
+  TransactionNumber: number;
+  Timestamp: string;
+  Signer: string;
+  Signature: string;
+  From: string;
+  To: string;
+  RelayerRecipientSOL: string;
+  Type: string;
+  PublicAmountSOL: number;
+  PublicAmountSPL: number;
+  RelayerFeeSOL: number;
+}
+class TransactionHistoryCommand extends Command {
+  static description = "Show user transaction history";
+  static flags = {
+    'skip-fetch': Flags.boolean({
+      char: "s",
+      description: "Retrieve the latest transaction history: skip fetching from the indexer",
+      parse: async () => false, 
+      allowNo: true,
+      hidden: true,
+      default: true,
+    }),
+  };
+
+  async run() {
+    const { flags } = await this.parse(TransactionHistoryCommand);
+    const latest = flags["skip-fetch"];
+
+    const loader = new CustomLoader("Retrieving user transaction history...");
+    loader.start();
+
+    try {
+      this.log('\n');
+
+      const user = await getUser();
+      const transactions: IndexedTransaction[] = await user.getTransactionHistory();
+
+      transactions.reverse().forEach((transaction, index) => {
+        let date = new Date(transaction.blockTime);
+        let transactionHistory: TransactionHistory = {
+          TransactionNumber: index,
+          Timestamp: date.toString(),
+          Type: `\x1b[32m${transaction.type}\x1b[0m`,
+          PublicAmountSOL: this.convertToSol(transaction.publicAmountSol),
+          PublicAmountSPL: transaction.publicAmountSpl / 100,
+          From: transaction.from.toString(),
+          To: transaction.to.toString(),
+          RelayerRecipientSOL: transaction.relayerRecipientSol.toString(),
+          RelayerFeeSOL: this.convertToSol(transaction.relayerFee),
+          Signer: transaction.signer.toString(),
+          Signature: transaction.signature,
+        };
+        switch (transaction.type) {
+          case "SHIELD":
```

Comment:
cc @ananas-block 

I don't consider it a blocker for this PR (that probably would require a more intrusive change in zk.js), but what do you think about changing that to enum? String comparisons are more expensive than int/enum comparisons.

Reply:
there is an enum called Action it is also string based though xD